### DISCUSSION TITLE
"WinGet malfuntion detected" I press repair WinGet and the same message keeps appearing.

### DISCUSSION THREAD

**Discussion by Inmortel** [Devolutions/UniGetUI#2902] [General]
![Sin título](https://github.com/user-attachments/assets/fe034138-400f-4e13-9ed6-9ca8170ea7f9)
When the process is finished, I get an advice that says it has been successfully repaired, I restart the program and the error continues to appear.
This happened after updating the software for the first time, I had 8 programs and among them the only one that failed to update Discord, when I opened the Discord app it updated, but now this error message appears in WinGet and nothing appears in the installed packages

**Comment by marticliment** [comment]
Which UniGetUI* version are you using?

**Reply by Inmortel** [reply]
where can i see my winget version?

**Reply by marticliment** [reply]
UniGetUI, sorry

**Reply by Inmortel** [reply]
3.1.3

**Reply by marticliment** [reply]
Please attach UniGetUI logs

**Reply by Inmortel** [reply]
![Sin título](https://github.com/user-attachments/assets/7ea600c2-c6d9-4a1a-bff9-1a674e61a72e)

**Reply by Inmortel** [reply]
[UniGetUI Log.txt](https://github.com/user-attachments/files/17548351/UniGetUI.Log.txt)

**Reply by marticliment** [reply]
Please go to UniGetUI Settings -> Package Managers -> WinGet -> Enable bundled WinGet instead of using the COM APIs

**Reply by Inmortel** [reply]
OK now works, thank you very much
Discord appears again with an update and still has an error when updating.
But now the installed packages reappeared.
Should I ignore Discord updates?

**Reply by Inmortel** [reply]
When the discord package update fails, the problem occurs again and the option you told me is unchecked, but if I check it again, it is fixed, it seems that discord is the one causing the problem.

**Reply by Inmortel** [reply]
Now that I read the option you told me, I'm starting to worry. 
My system version of WinGet is incorrect or corrupted, should I reinstall it or repair it in some way?

**Reply by marticliment** [reply]
Not necessarily corrupted, but it is not being able to properly fetch packages via COM. Please just ignore discord updates for now, and I will investigate why this is happening and will attempt to fix it

**Reply by GitHub-Enrico** [reply]
> Please go to UniGetUI Settings -> Package Managers -> WinGet -> Enable bundled WinGet instead of using the COM APIs

For me it didn't work.
On 3.15 version, "WinGet malfuntion detected" is still appearing.

**Comment by alexolog** [comment]
Same here, can't get rid of it

**Comment by Inmortel** [comment]
I recently switched to Win 11 and the app now works correctly.

**Comment by maspiers** [comment]
I've recently started getting this same error on the previous version and now on version 3.1.7, running on Windows 11 24H2.
Looking in the log file, there's an error in yellow:
Manager WinGet failed to list available updates with exception InvalidOperationException: WinGet: Failed to connect to composite catalog.
followed by a lot more errors in red.
[UniGetUI Log 20250303.txt](https://github.com/user-attachments/files/19056033/UniGetUI.Log.20250303.txt)

**Reply by marticliment** [reply]
does the WinGet command work? (example `winget list`)

**Comment by maspiers** [comment]
Yes.

**Comment by foinf** [comment]
This error appears if you open UnigetUI and are not connected to the internet. Pretty misleading error if that's the case.

**Reply by marticliment** [reply]
This shouldn't be the case anymore... Which UniGetUI version are you running?

**Reply by foinf** [reply]
> This shouldn't be the case anymore... Which UniGetUI version are you running?

Hi!

I am using 3.2.0
Winget version v1.10.390
Windows 11 Pro 24H2

Thanks.

**Reply by marticliment** [reply]
I will investigate

**Reply by Knud3** [reply]
I had that exact problem when Pi-hole was blocking certain DNS requests. WinGet was otherwise working without problem.

**Reply by foinf** [reply]
> I had that exact problem when Pi-hole was blocking certain DNS requests. WinGet was otherwise working without problem.

I am not using any ad blocking feature and it does happen still.
